### PR TITLE
Handle network hooks located in /rw/config/network-hooks.d

### PR DESCRIPTION
**Pull request by fepitre**
Example:
/rw/config/network-hooks.d/test.sh
\#!/bin/bash

command="$1"
vif="$2"
ip="$3"

if [ "$ip" == '10.137.0.100' ]; then
    case "$command" in
        online)
            ip route add 192.168.0.100 via 10.137.0.100
            ;;
        offline)
            ip route del 192.168.0.100
            ;;
    esac
fi

### CODE REVIEW

**Comment by fepitre** [comment]
I also cleanup the tabs by replacing them with spaces.

**Comment by marmarek** [comment]
As for metric, maybe just export that variable. I don't think it should be part of an API we'll need to keep stable.

File: network/vif-route-qubes [hunk 1]
```diff
@@ -95,19 +95,29 @@ domid=${domid/.*/}
...
         echo -e "*raw\\n$iptables_cmd -i ${vif} ! -s ${addr} -j DROP\\nCOMMIT" | \
             ${cmdprefix} $ipt --noflush $ipt_arg
-	done
+
+        # Network Hooks for triggering supplementary actions on AppVM connect
+        if [ -d /rw/config/network-hooks.d ]; then
+            for hook in /rw/config/network-hooks.d/*
+            do
+                log debug "Executing network-hook $(basename "$hook")..."
+                do_without_error "${hook}" "${command}" "${vif}" "${addr}"
```
Comment (marmarek) [inline network/vif-route-qubes]: Check if file is executable first.

File: network/vif-route-qubes [hunk 2]
```diff
@@ -120,8 +130,7 @@ if [ "${ip}" ] ; then
 fi
```
Comment (marmarek) [inline network/vif-route-qubes]: IMO the hook should be called even if IP isn't specified. In that case, the argument should be empty.

**Comment by marmarek** [comment]
Also, please add info to https://www.qubes-os.org/doc/config-files/, including parameters, when hooks are called etc.

**Pull request merged**